test/integration_test/kubevirt_test.go

```diff
@@ -246,3 +251,67 @@ func createDatadiskTemplates(t *testing.T) error {
 	}
 	return nil
 }
+
+func kubevirtVMScaledDeployAndValidate(t *testing.T, instanceID, appKey string, scale int) []*scheduler.Context {
+	var wg sync.WaitGroup
+	var allCtxs []*scheduler.Context
+	logrus.Infof("Deploying %d VMs and validating them.", scale)
+	for i := 1; i <= scale; i++ {
+		logrus.Infof("Deploying VM: %d", i)
+		wg.Add(1)
+		ctxs := kubevirtVMsDeployAndValidate(
+			t,
+			instanceID+"-"+fmt.Sprintf("%d", i),
+			[]string{appKey},
+			&wg,
+		)
+		logrus.Infof("Validated VM: %d", i)
+		allCtxs = append(allCtxs, ctxs...)
+	}
+	wg.Wait()
```

Comment:
I am wondering if sync.WaitGroup is an overkill here. WaitGroup is needed when multiple goroutines are working in parallel and you want all of them to finish before proceeding. But I am not seeing any concurrent work being done here. Am I missing something?

Reply:
I had go routines earlier but forgot to remove the sync group. Fixed.